tests/ha/upgrade_from_sle11sp4_workarounds.pm

```diff
@@ -68,6 +68,9 @@ sub run {
     systemctl 'enable sbd';
     systemctl 'enable pacemaker';
 
+    # Start iscsi service if it is not started yet
+    systemctl 'start iscsi' if (systemctl 'status iscsi', ignore_failure => 1);
```

Comment:
As this seems to be sporadic, I supposed that the `iscsi` service is enabled and so this looks like a bug... Maybe a bsc# should be opened and a soft-failure added in the `if` condition, WDYT?

Reply:
From what we could see while debugging yesterday, I'm thinking on any of two options:

1. iscsi service does not start as it should, hence bug.
2. iscsi service does start, but the support server is not yet started and the client fails with connection timeout. If this is the case, the best workaround would be to sync all nodes with the support server once we are sure that the support server has started the iscsi server, but this is already being done with `wait_barrier`, so any extra sync between jobs would require some re-writes to the support server code for migrations.

The error messages linked by @juadk point to option 2, but that does not explain why we're only seeing this in 11-SP4 and not in the other upgrade scenarios, unless this is being affected by the cluster not starting automatically when migrating from 11-SP4.

Reply:
Yes, I think we are in the option 2 that you mentioned. I really think we are facing to a race condition rather than a bug.

Reply:
I still continue to think that grub timeout is a valid option...
I checked [migration from 11sp4 to 12sp5]( https://openqa.suse.de/tests/3577273#step/boot_to_desktop/1) and there is no grub timeout there. 
In the other hand, we can see it for [migration from 11sp4 to 15sp2](https://openqa.suse.de/tests/3717856#step/boot_to_desktop/1).
From my understanding, next wait_barriers doesn't prevent iscsi starting during boot time if we have a timeout in grub.
@alvarocarvajald @ldevulder WDYT?

Reply:
Yes in that case when the timeout is reached the OS is booted, so iSCSI client tries to start before support-server is completed. The `barrier_wait` here is just to pause the test itself, not the timeout. This is why there should be no timeout here, in that case `Enter` will be pressed after the `barrier_wait`, and so the OS will be booted after support-server is completed.

@juadk is the 11sp4 images have re-created recently? Maybe the timeout is here? As I don't have this issue with sles4sap15-sp2 I'm not sure that this is a product bug, as it's the same Installer, but it's a possibility.

Reply:
> I still continue to think that grub timeout is a valid option...
> I checked [migration from 11sp4 to 12sp5](https://openqa.suse.de/tests/3577273#step/boot_to_desktop/1) and there is no grub timeout there.
> In the other hand, we can see it for [migration from 11sp4 to 15sp2](https://openqa.suse.de/tests/3717856#step/boot_to_desktop/1).
> From my understanding, next wait_barriers doesn't prevent iscsi starting during boot time if we have a timeout in grub.
> @alvarocarvajald @ldevulder WDYT?

OK. I think I have the culprit for that. I removed `installation/disable_grub_timeout` from the offline migration tests as I assumed it's not needed (it's not needed for the other qcow2 images), but it seems it is required for the 11-SP4 tests. We can try re-enabling it, but IIRC this module has problems in the offline installation with VIDEOMODE=text.

However, this grub timeout is in the cluster nodes. The one we need to start faster is the support server which is not affected by the offline migration tests, and does have a grub timeout in the 12-SP5 migration scenario you linked (https://openqa.suse.de/tests/3577270#step/login/1), however, there it works fine.

Reply:
Tests re-triggered without `disable_grub_timeout` in  `EXCLUDE_MODULES`, then we will decide what to do :smiley:

Reply:
> We can try re-enabling it, but IIRC this module has problems in the offline installation with VIDEOMODE=text.

Which issue do you have in textmode? Looks fine in sles4sap: https://openqa.suse.de/tests/3722780#step/disable_grub_timeout/9

Anyway I'm personally not a big fan of excluding failing modules, but I don't know what was the issue here.

> However, this grub timeout is in the cluster nodes. The one we need to start faster is the support server which is not affected by the offline migration tests, and does have a grub timeout in the 12-SP5 migration scenario you linked (https://openqa.suse.de/tests/3577270#step/login/1), but it works fine there.

Agree that support-server should be started faster and first, so with the Grub timeout activated here the cluster nodes are automatically started *AFTER* the timeout (8s) so faster than the support-server, and the `barrier_wait` is in fact done after the boot of the nodes. By deactivating the Grub timeout we force the nodes to stay in Grub, wait with `barrier_wait` and then boot the OS, so after the support-server.

Reply:
> Which issue do you have in textmode? Looks fine in sles4sap: https://openqa.suse.de/tests/3722780#step/disable_grub_timeout/9
> 

Looks fine in installation. In HA there are also some of those: https://openqa.suse.de/tests/3715205#step/disable_grub_timeout/9 (for example)

The problem is in migrations I believe.

Issue is simple, but requires changes to the code and since the qcow2 images themselves get the grub timeout disabled during the original installation, I decided this was not needed. Problem is the 11-SP4 qcow2 images weren't generated by openQA, but by hand.

WRT the issue: https://openqa.suse.de/tests/3686302#step/disable_grub_timeout/3

It's failing there due to a missing needle, but even if we create the needle, code needs to be changed as after the needle it sends an `alt-b` to get to the bootloader configuration screen, but in these migrations `alt-b` goes to **B**ackup.

> Anyway I'm personally not a big fan of excluding failing modules, but I don't know what was the issue here.

Me neither, but as I say, this was a simpler and quicker solution given the problem. Now that there's an issue that seems to be caused by the removal of that module, there's a reason to go for the more complicated route.

> Agree that support-server should be started faster and first, so with the Grub timeout activated here the cluster nodes are automatically started _AFTER_ the timeout (8s) so faster than the support-server, and the `barrier_wait` is in fact done after the boot of the nodes. By deactivating the Grub timeout we force the nodes to stay in Grub, wait with `barrier_wait` and then boot the OS, so after the support-server.

Makes sense. Good catch.

Reply:
Let's closed this PR, I will create needles and change code next week. Thanks for your analyze.